bob-access/src/credentials.rs

```diff
@@ -20,23 +20,25 @@ pub struct CredentialsHolder<A: Authenticator> {
     pd: PhantomData<A>,
 }
 
-impl<A: Authenticator> CredentialsHolder<A> {
-    pub fn into_credentials(self) -> Credentials {
-        self.credentials
+impl<A: Authenticator> From<CredentialsHolder<A>> for Credentials {
+    fn from(holder: CredentialsHolder<A>) -> Self {
+        holder.credentials
     }
 }
 
 #[derive(Debug, Clone)]
 pub enum CredentialsKind {
     Basic { username: String, password: String },
     Token(String),
+    InterNode(String),
 }
 
 #[derive(Clone, Copy, PartialEq, Debug, Serialize, Deserialize)]
 pub enum CredentialsType {
     None,
     Basic,
     Token,
+    InterNode,
```

Comment:
`InterNode` is not needed in this enum. This is basically an `AuthenticationType`, so I suggest to rename it and move closer to `Authenticator`